AlphaWallet/Market/Coordinators/UniversalLinkCoordinator.swift

```diff
@@ -156,19 +156,35 @@ class UniversalLinkCoordinator: Coordinator {
 
     func completeOrderHandling(signedOrder: SignedOrder)
     {
+        let requiresPaymaster = requiresPaymasterForCurrencyLinks(signedOrder: signedOrder)
         if signedOrder.order.price == 0 {
             self.checkPaymentServerSupportsContract(contractAddress: signedOrder.order.contractAddress) { supported in
```

Comment:
Here's a version using guard to make it clearer that is the precondition you are looking out for:

```
            self.checkPaymentServerSupportsContract(contractAddress: signedOrder.order.contractAddress) { supported in
                //Currency links on mainnet/classic/xdai without a paymaster should be rejected for security reasons (front running)
                guard supported || !requiresPaymaster else {
                    self.showImportError(errorMessage: R.string.localizable.aClaimTokenFailedServerDown())
                    return
                }

                if supported {
                    self.usePaymentServerForFreeTransferLinks(signedOrder: signedOrder)
                } else {
                    self.handlePaidImports(signedOrder: signedOrder)
                }
            }
```

Checking against classic means we don't allow classic drops until payment server supports them right?

Reply:
Any network with value should only allow paymaster